src/main.rs

```diff
@@ -638,40 +681,94 @@ mod test {
                 "failed to clear test data, the error is: {}",
                 clear_res.unwrap_err(),
             );
-
-            let (csi_server, worker_server) = match build_grpc_servers(
-                end_point,
-                worker_port,
-                node_id,
-                driver_name,
-                data_dir,
-                run_as,
-                etcd_client,
+            let meta_data =
+                match build_meta_data(worker_port, node_id, data_dir, run_as, etcd_client) {
+                    Ok(md) => md,
+                    Err(e) => panic!("failed to build meta data, the error is : {}", e,),
+                };
+            let md = Arc::new(meta_data);
+            let controller_server = match build_grpc_controller_servers(
+                &end_point,
+                &driver_name,
+                Arc::<MetaData>::clone(&md),
             ) {
-                Ok((s1, s2)) => (s1, s2),
-                Err(e) => panic!(
-                    "failed to build CSI server and worker server, \
-                        the error is : {}",
-                    e,
-                ),
+                Ok(server) => server,
+                Err(e) => panic!("failed to build CSI server, the error is : {}", e,),
             };
+            let node_server = match build_grpc_node_servers(
+                &node_end_point,
+                &driver_name,
+                Arc::<MetaData>::clone(&md),
+            ) {
+                Ok(server) => server,
+                Err(e) => panic!("failed to build Node server, the error is : {}", e,),
+            };
+            let worker_server = match build_grpc_worker_servers(worker_port, md) {
+                Ok(server) => server,
+                Err(e) => panic!("failed to build Worker server, the error is : {}", e,),
+            };
+
             // Keep running the task in the background
             let _th = thread::spawn(move || {
                 if async_server {
-                    run_async_servers(csi_server, worker_server);
+                    run_async_controller_and_node_servers(
```

Comment:
you don't need to create `run_async_controller_and_node_servers()`, you can reuse `run_async_controller_server()` and `run_async_node_server()` by running them in different thread respectively

Reply:
Yes, we can, but we need to spawn a new thread. I think have run_async_controller_and_node_servers is more readable and bring less overhead.

Reply:
spawn a new thread here won't take too much resource here, furthermore reuse run_async_controller_server() and run_async_node_server() will test them too